src/keys.rs

```diff
@@ -133,11 +137,25 @@ impl PartialEq for SpendValidatingKey {
     }
 }
 
+impl Eq for SpendValidatingKey {}
+
 impl SpendValidatingKey {
     /// Randomizes this spend validating key with the given `randomizer`.
     pub fn randomize(&self, randomizer: &pallas::Scalar) -> redpallas::VerificationKey<SpendAuth> {
         self.0.randomize(randomizer)
     }
+
+    /// Converts this spend validating key to its serialized form.
+    pub(crate) fn to_bytes(&self) -> [u8; 32] {
+        <[u8; 32]>::from(&self.0)
```

Comment:
I think this is serializing ak<sup>P</sup> (i.e. the compressed point), not ak (the x-coordinate, of type P<sub>x</sub>).